### ISSUE TITLE
Line numbers for terraform parsing errors?

### ISSUE THREAD

**Issue by Andrew-Poloni** [hashicorp/terraform#16665]
Gentlemen,

A feature request: in the case where Terraform encounters a parsing error in a Terraform file, would it be possible to print the line number in the Terraform file where the parsing error was encountered?

Many thanks in advance!

**Comment by apparentlymart** [comment]
Hi @Andrew-Poloni!

We are currently working on a new language parser that will make better error output possible. It will take a few iterations before all messages are updated, but it's definitely a goal.

Thanks for this suggestion!

**Comment by Andrew-Poloni** [comment]
Martin,

That sounds great! :) It's sometimes a bit of a challenge finding the exact source of a parsing issue ... thank you! :+1:

**Comment by EUCConsulting** [comment]
+1 This would really make troubleshooting easier

**Comment by shapeofarchitect** [comment]
+1 This is really needed. The below error just doesn't help.
`: parse error at 2:92: expected "}" but found "."`

**Comment by apparentlymart** [comment]
Hi @shapeofarchitect!

In that case I think the message is telling you that the problem is on line 2, column 92 of one of your configuration files. Unfortunately the current parser does not remember which file each token came from, so it can't tell you the filename. This is one of the limitations fixed in the new parser, which we're still working on integrating and plan to include in the next major release.

Here's a screenshot of an error message from the new version:

![Error: unsupported attribute at try.tf line 7, in variable "aws_region"](https://user-images.githubusercontent.com/20180/37983515-8e6a21ea-31a8-11e8-9d22-041dcadd6d46.png)

For readers who can't see images, here's the same content as text:

```
$ terraform validate

Error: Unsupported attribute

  on try.tf line 7, in variable "aws_region":
   7:   defualt = "us-west-2"

An attribute named "defualt" is not expected here. Did you mean "default"?
```

**Comment by shapeofarchitect** [comment]
@apparentlymart  much thanks yes by those numbers I meant more detailed function or attribute resource, New Parser looks great as I hope we would have more detailed error reports. Great work there , can't wait to use :)

**Comment by apparentlymart** [comment]
Hi all!

I'm pleased to report that the new error message system I mentioned earlier has now been merged into master ready to be included in the forthcoming v0.12.0 final release.

Since this is now merged, I'm going to close out this issue. Thanks for this request, and thanks for your patience while we laid the groundwork to make it possible.
